contributed/DC2_object_deredden.ipynb

```diff
@@ -0,0 +1,369 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# DC2 object catalog: removing Milky Way dust\n",
+    "\n",
+    "Contributed by: **Sam Schmidt [@sschmidt23](https://github.com/LSSTDESC/DC2-analysis/issues/new?body=@sschmidt23)**\n",
+    "\n",
+    "Last Verifed to Run: **2020-06-19** (by @sschmidt23)\n",
+    "\n",
+    "The DC2 object catalogs generated from mock images have simulted Milky Way dust included.  If you need access to object colors, effects of this foreground must be removed.  This notebook will give a very quick demo for using the `dustmaps` package to remove Milky Way foreground dust from the DC2 object catalogs.\n",
+    "\n",
+    "Typically, foreground dust is parameterized by E(B-V), and the amount of dereddening in each specific band, A_lambda, is found via an A_lambda/E(B-V) parameter specific to each filter.  A separate notebook [LINK HERE LATER] can show you how to derive these parameters for the LSST bandpass shapes assumed in the DC2 simulations, but for the purposes of this notebook we will simply list the A_lambda/E(B-V) parameters, which for filters `u,g,r,i,z,y` are `4.81,3.64,2.70,2.06,1.58,1.31`\n",
+    "\n",
+    "__Logistics__: This notebook is intended to be run through the Jupyter Lab NERSC interface available here: https://jupyter.nersc.gov/. To setup your NERSC environment, please follow the instructions available here: https://confluence.slac.stanford.edu/display/LSSTDESC/Using+Jupyter+at+NERSC\n",
+    "\n",
+    "__Other notes__: \n",
+    "This demo uses the non-DESC `dustmaps` package, which employs astropy units, so both of these packages must be available in the path of the user.  Note that both packages are available in the `desc-python-dev` kernel"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The DC2 simulations assume SFD reddening with interpolation between the pixels set. the `dustmaps` package can work with several dust maps derived from a variety of sources.  We will point the dustmaps code to the SFD maps with the config['data_dir'] parameter in the cell below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import GCRCatalogs\n",
+    "import dustmaps\n",
+    "from dustmaps.sfd import SFDQuery\n",
+    "from astropy.coordinates import SkyCoord\n",
+    "from dustmaps.config import config\n",
+    "import matplotlib.pyplot as plt\n",
+    "config['data_dir'] = '/global/cfs/cdirs/lsst/groups/PZ/PhotoZDC2/run2.2i_dr6_test/TESTDUST/mapdata' #update this path when dustmaps are copied to a more stable location!\n",
```

Comment:
Should we move this file to the `shared` directory?

Reply:
I think so, it was still an open question as to whether to use the `dustmaps` package when I made this, but I think it's clearly a nice solution, so copying the two files to a location on shared would be best.  The files are actually pretty small, I think 2x 50 MB. 
If the path is updated I can modify the notebook to point to the new path.